src/migration/tasks/MigrateUserSettingsTask.ts

```diff
@@ -0,0 +1,45 @@
+import { UserSetting, UserSettingsContentType, UserSettingsType } from '../../types/Setting';
+
+import Constants from '../../utils/Constants';
+import MigrationTask from '../MigrationTask';
+import SettingStorage from '../../storage/mongodb/SettingStorage';
+import Tenant from '../../types/Tenant';
+import TenantStorage from '../../storage/mongodb/TenantStorage';
+import Utils from '../../utils/Utils';
+
+export default class MigrateUserSettingsTask extends MigrationTask {
+  public async migrate(): Promise<void> {
+    const tenants = await TenantStorage.getTenants({}, Constants.DB_PARAMS_MAX_LIMIT);
+    for (const tenant of tenants.result) {
+      await this.migrateTenant(tenant);
+    }
+  }
+
+  public async migrateTenant(tenant: Tenant): Promise<void> {
+    const userSetting = await SettingStorage.getUserSettings(tenant.id);
+    // If no user setting exists, initialize it
+    if (Utils.isEmptyObject(userSetting)) {
+      // Create new user setting with account activation param
+      const settingsToSave = {
+        identifier: UserSettingsType.USER,
+        content: {
+          type: UserSettingsContentType.ACCOUNT_ACTIVATION,
+          accountActivation: {
+            doNotActivateByDefault: false
```

Comment:
I'm not very fond of that name: `enabled: true/false` is enough

Reply:
I agree, I didn't find something clear enough and forgot to ask before PR @LucasBrazi06 is 'enabled' ok ?

Reply:
If we follow the same Settings structure we should have something like this:
```
{ 
    "_id" : ObjectId("5c6c8e8ee7fd060008215e30"), 
    "identifier" : "user", 
    "content" : {
        "type" : "user", 
        "user" : {
            "doNotActivateByDefault": true
        }
    }, 
    "createdBy" : null, 
    "createdOn" : ISODate("2019-02-19T23:17:34.189+0000"), 
    "lastChangedBy" : null, 
    "lastChangedOn" : ISODate("2019-06-14T09:38:08.048+0000"), 
    "sensitiveData" : []
}
```
@cabch I thought this was already discussed last Thursday ;-)

Reply:
>     "content" : {
>         "type" : "user", 
>         "user" : {
>             "doNotActivateByDefault": true
>         }

accountActivationByDefault: true/false is more understandable.

No other settings data structure use a negation for feature enablement on boolean. From a user experience perpective, it makes no sense: 

export interface BillingSetting {
```
  usersLastSynchronizedOn?: Date;
  invoicesLastSynchronizedOn?: Date;
}

export interface StripeBillingSetting extends BillingSetting {
  url: string;
  secretKey: string;
  publicKey: string;
  noCardAllowed: boolean;
  immediateBillingAllowed: boolean;
  periodicBillingAllowed: boolean;
  advanceBillingAllowed: boolean;
  currency: string;
  taxID: string;
}
```

Reply:
> No other settings data structure use a negation for feature enablement on boolean. From a user experience perpective, it makes no sense:

I agree.
@cabch Could you please rename doNotActivateByDefault to autoActivateAccountAfterValidation or something equivalent and proceed with the changes in the front-end also?
Thanks.

Reply:
replaced manualAccountActivation to autoActivateAccountAfterValidation and all the true/false resulting from it

Reply:
true by default (= users are auto activated for every new tenants and for all the existing tenants so the behaviour doesn't change from today's)